exams/pearson/download.py

```diff
@@ -50,6 +56,37 @@ def locally_extracted(zip_file, member):
             os.remove(extracted_file_path)
 
 
+def format_and_log_error(message, error=None, client_candidate_id=None,
+                         client_authorization_id=None, username=None, **kwargs):
+    """
+    Formats and logs an error message
+
+    Args:
+        error: error message
+        client_candidate_id: the candidate id (Profile.student_id on our end) returned in the result
+        client_authorization_id: the authorization id (ExamAuthorization.id on our end) returned in the result
+        username: User's username
+
+    Returns:
+        str: formatted error message
+    """
+    formatted_message = ''.join([
+        message,
+        ' authorization_id=`{authorization_id}`' if client_authorization_id else '',
+        ' student_id=`{student_id}`' if client_candidate_id else '',
+        ' username=`{username}`' if username else '',
+        ' error=`{error}`' if error else '',
+    ]).format(
+        authorization_id=client_authorization_id,
+        student_id=client_candidate_id,
+        username=username,
+        error=error,
+        **kwargs
+    )
```

Comment:
Here's a possibly cleaner and probably-not-working way to rewrite this function:
 
```python
value_output_dict = dict(
    authorization_id=client_authorization_id,
    student_id=client_candidate_id,
    username=username,
    error=error
)
value_output = ' '.join(['{}={}'.format(k, v) for k, v in value_output_dict.items() if v])
formatted_message = ' '.join([message, value_output]).format(**kwargs)
log.error(formatted_message)
return formatted_message
```